File: backend/model/RoommateGroup.js
const mongoose = require("mongoose");

const RoommateGroupSchema = new mongoose.Schema({
  name: { type: String, required: true },
  owner_id: { type: mongoose.Schema.Types.ObjectId, ref: "User", required: true },
  roommates: [{ type: mongoose.Schema.Types.ObjectId, ref: "User" }],
  tasks: [{ type: mongoose.Schema.Types.ObjectId, ref: "Task" }],
  bills: [{ type: mongoose.Schema.Types.ObjectId, ref: "Bill" }],
  grocery_items: [{ type: mongoose.Schema.Types.ObjectId, ref: "GroceryItem" }],
  shared_resource: [{ type: mongoose.Schema.Types.ObjectId, ref: "SharedResource" }]
});

module.exports = mongoose.model("RoommateGroup", RoommateGroupSchema);

File: App/Roommate App/Models/RoommateGroup.swift
//
//  RoommateGroup.swift
//  Roommate App
//
//  Created by Logan Norman on 4/5/23.
//

import Foundation

class RoommateGroup {
    let name: String
    let roommates: [User]
    let bills: [Bill]
    let tasks: [GroupTask]
    let groceries: [GroceryItem]
    
    init(name: String, roommates: [User], bills: [Bill]? = nil, tasks: [GroupTask]? = nil, groceries: [GroceryItem]? = nil) {
        self.name = name
        self.roommates = roommates
        self.bills = bills ?? []
        self.tasks = tasks ?? []
        self.groceries = groceries ?? []
    }
}
